### PR TITLE
Refactor report header

### PR DESCRIPTION
**Pull request by ayeswarya**
Resolves #

## Description
Refactor report header

## Screenshots
Screen Shot 2021-03-09 at 6 33 35 PM
Screen Shot 2021-03-09 at 6 33 59 PM
Screen Shot 2021-03-09 at 6 34 09 PM

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/report-header.hbs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+  <div class="container-full m-b-4">
```
Comment (jkusa) [inline packages/reports/addon/components/report-header.hbs:1]: is this indented

Comment (jkusa) [reply packages/reports/addon/components/report-header.hbs:1]: also ©️

File: packages/reports/addon/components/report-header.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+
+import Component from '@glimmer/component';
+import ReportModel from 'navi-core/models/report';
+
+interface ReportHeaderArgs {
+  model: ReportModel;
+  addToDashboard: (reportId: string, reportName: string) => void;
+  addToNewDashboard: (dashboardTitle: string, reportName: string) => void;
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/report-header.ts:12]:
```suggestion
  addToDashboard?: (reportId: string, reportName: string) => void;
  addToNewDashboard?: (dashboardTitle: string, reportName: string) => void;
```

**Review by kevinhinterlong — COMMENTED** [review]
report header

Some issues I noticed playing around 
- Tooltips are not showing up correctly (it seems like several render at the same time)
- Lost vertical spacing on report header
- Make sure to set feature flag `enableScheduleReports` to true so you can fix the schedule button styles
- Can we make sure the actions wrap around (add a row class to `navi-report__header`)

File: packages/reports/addon/components/report-header.hbs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+  <div class="container-full" ...attributes>
```
[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/addon/components/report-header.hbs:1]: Also, it looks like line 1 is indented, so the whole file needs to shift to the left

File: packages/reports/addon/components/report-header.hbs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    <div class="row">
+      <div class="col-1-1">
+        {{#if (feature-flag "enableDirectory")}}
+          <LinkTo @route="directory.my-data" class="navi-report__breadcrumb-link">Directory</LinkTo>
```
Comment (jkusa) [inline packages/reports/addon/components/report-header.hbs:5]: Can we change all the `navi-report__` to `report-header__`

**Comment by jkusa** [comment]
Looks like we have two button rows here:
Screen Shot 2021-03-12 at 3 29 45 PM

**Comment by ayeswarya** [comment]
> Looks like we have two button rows here:
> Screen Shot 2021-03-12 at 3 29 45 PM

I have made changes to rectify this.

File: packages/reports/app/styles/navi-reports/common/report.scss [hunk 5]
```diff
@@ -9,33 +9,29 @@
...
+    }
 
-  &__header {
-    align-items: center;
-    display: flex;
-    justify-content: space-between;
-    margin-bottom: 10px;
-    min-height: min-content;
+    &__header-action {
+      display: flex;
```
Comment (jkusa) [inline packages/reports/app/styles/navi-reports/common/report.scss:27]:
```suggestion
      display: flex;
      button {
        white-space: nowrap;
      } 
```
This ^^ should fix this: 

Screen Shot 2021-03-16 at 7 11 39 PM

**Review by jkusa — COMMENTED** [review]
Screen Shot 2021-03-16 at 7 16 57 PM
Seems like the tooltip are off

**Comment by ayeswarya** [comment]
> Seems like the tooltip are off
Screen Shot 2021-03-17 at 3.32.14 PM
Screen Shot 2021-03-17 at 3 32 19 PM
Screen Shot 2021-03-17 at 3 32 25 PM
Screen Shot 2021-03-17 at 3 32 30 PM
Screen Shot 2021-03-17 at 3 32 35 PM

File: packages/reports/addon/templates/components/report-actions.hbs [hunk 6]
```diff
@@ -2,128 +2,122 @@
...
-          disabled={{not @model.request.validations.isTruelyValid}}
-          {{on "click" onOpen}}
-        >
-          <NaviIcon @icon="clock-o" class="navi-report__action-icon" />
-          <span>Schedule</span>
-        </button>
-      </CommonActions::Schedule>
-      <EmberTooltip @text="Schedule a report run" />
-    </li>
+        <NaviIcon @icon="clock-o" class="navi-report__action-icon" />
```
Comment (kevinhinterlong) [inline packages/reports/addon/templates/components/report-actions.hbs:105]:
```suggestion
        <NaviIcon @icon="clock-o" class="navi-report__action-icon m-r-6" />
```

**Review by kevinhinterlong — COMMENTED** [review]
report header

Looks like export needs
```css
{
  text-align: center;
  display: inline-flex;
}
```

and some margin for schedule icon

File: packages/reports/addon/templates/components/report-actions.hbs [hunk 7]
```diff
@@ -2,128 +2,122 @@
...
-          <NaviIcon @icon="download" class="navi-report__action-icon" />
+          <NaviIcon @icon="download" class="m-r-6" />
           Export
         </ExportAction>
       {{/let}}
     {{/let}}
-    <EmberTooltip>
-      {{if @model.request.validations.isTruelyValid "Export the report" "Run a valid report to enable export"}}
+    <EmberTooltip @targetId="report-action__export">
+      {{if @model.isNew "Save report to enable clone" "Clone the report"}}
```
Comment (kevinhinterlong) [inline packages/reports/addon/templates/components/report-actions.hbs:68]:
```suggestion
      {{if @model.isNew "Save report to enable export" "Export the report"}}
```

Comment (ayeswarya) [reply packages/reports/addon/templates/components/report-actions.hbs]: Thanks for catching that

File: packages/reports/addon/templates/components/report-actions.hbs (hunk 6 above)
Comment (ayeswarya) [reply packages/reports/addon/templates/components/report-actions.hbs]: Made these changes

**Review by jkusa — APPROVED** [review]

**Pull request merged**